homeassistant/components/tts/__init__.py

```diff
@@ -137,15 +137,16 @@ def async_default_engine(hass: HomeAssistant) -> str | None:
     component: EntityComponent[TextToSpeechEntity] = hass.data[DOMAIN]
     manager: SpeechManager = hass.data[DATA_TTS_MANAGER]
 
-    if "cloud" in manager.providers:
-        return "cloud"
+    default_entity_id: str | None = None
 
-    entity = next(iter(component.entities), None)
+    for entity in component.entities:
+        if entity.platform and entity.platform.platform_name == "cloud":
+            return entity.entity_id
 
-    if entity is not None:
-        return entity.entity_id
+        if default_entity_id is None:
```

Comment:
This is always true. Seems like the logic isn't correct.

Reply:
Nevermond. It's indented.

Reply:
I also thought the code looked wrong at first, but here's how it works, I think it's correct:

- Loop over all entities
  - If we find a cloud entity, return it
  - If this is the first non cloud entity, assign it to `default_entity_id`
  
The condition "if this is the first entity" makes the behavior more similar to the behavior without this PR: We fall back to the *first* non cloud entity